### PR TITLE
OCPBUGS-2316: Mixing ICMP v4 and v6 config causes a panic

### PR DESCRIPTION
**Pull request by msherif1234**
webhhok ICMP and ICMPv6 check was missing
for more details refer to issue #205
manual e2e run results

```
JUnit report was created: /tmp/test_e2e_logs/e2e_junit.xml

Ran 11 of 11 Specs in 168.696 seconds
SUCCESS! -- 11 Passed | 0 Failed | 0 Pending | 0 Skipped
```

Signed-off-by: Mohamed Mahmoud <mmahmoud@redhat.com>

### CODE REVIEW

File: api/v1alpha1/ingressnodefirewall_types.go [hunk 1]
```diff
@@ -57,23 +62,23 @@ type IngressNodeProtocolConfig struct {
 	Protocol IngressNodeFirewallRuleProtocolType `json:"protocol"`
 
 	// tcp defines an ingress node firewall rule for TCP protocol.
-	// +optional
+	// +unionMember,optional
```
Comment (JoelSpeed) [inline api/v1alpha1/ingressnodefirewall_types.go:65]: These should be on separate lines, with this you've made your union member optional when the protocol is set

File: api/v1alpha1/ingressnodefirewall_types.go [hunk 2]
```diff
@@ -49,6 +49,11 @@ type IngressNodeFirewallProtoRule struct {
 
 // IngressNodeProtocolConfig is a discriminated union of protocol's specific configuration.
 // +union
+// +kubebuilder:validation:XValidation:rule="has(self.protocol) && self.protocol == 'TCP' ?  has(self.tcp) : !has(self.tcp)",message="tcp is required when protocol is TCP, and forbidden otherwise"
+// +kubebuilder:validation:XValidation:rule="has(self.protocol) && self.protocol == 'UDP' ?  has(self.udp) : !has(self.udp)",message="udp is required when protocol is UDP, and forbidden otherwise"
```
Comment (martinkennelly) [inline api/v1alpha1/ingressnodefirewall_types.go:53]: nice :)

File: bundle/manifests/ingressnodefirewall.openshift.io_ingressnodefirewalls.yaml [hunk 3]
```diff
@@ -182,6 +182,27 @@ spec:
                             required:
                             - protocol
                             type: object
+                            x-kubernetes-validations:
```
Comment (martinkennelly) [inline bundle/manifests/ingressnodefirewall.openshift.io_ingressnodefirewalls.yaml:185]: was this auto generated by kube builder?

**Review by martinkennelly — COMMENTED** [review]
LGTM

File: bundle/manifests/ingressnodefirewall.openshift.io_ingressnodefirewalls.yaml (hunk 3 above)
Comment (msherif1234) [reply bundle/manifests/ingressnodefirewall.openshift.io_ingressnodefirewalls.yaml:185]: yes all manifests and bundle are results of `make manifests` and `make bundle`

**Pull request merged**